src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -118,6 +128,9 @@ directory, relative to the root.",
                 .arg(Arg::with_name("root").long("root").required(true).takes_value(true).help(
                   "Root under which the globs live. The Directory proto produced will be relative \
 to this directory.",
+            ))
+                .arg(Arg::with_name("json").long("json").multiple(false).takes_value(true).help(
```

Comment:
Assuming that the default behaviour is to create boolean flags, it seems like this should not be `takes_value(true)`? Then if you specify the flag as `--json`, you get json.

But if it is going to take a value, something would need to actually happen to parse the boolean flag string into a value... `if let Some("true") = args.value_of("json")` will fail for some obvious cases.

Alternatively, boolean flags (particularly for "output mode") can be a bit limiting. So another alternative might be `--output-mode=(json|simple)` or something.